.github/workflows/golang-binary-test.yml

```diff
@@ -0,0 +1,76 @@
+name: Monorepo CI/CD
+
+on:
+  push:
+    paths-ignore:
+      - 'services/excluded/**/*'
+    paths:
+      - 'persys-cloud/**'
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Build and Deploy
+      run: |
+        # Determine the changed directory
+        CHANGED_DIR=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep '^persys-cloud/' | sed -E 's|^services/([^/]+).*|\1|' | uniq)
+
+        # Check if the changed directory is not in the excluded list
+        if [[ ! "$CHANGED_DIR" =~ ^(excluded|another_excluded_directory)$ ]]; then
```

Comment:
considering that we are excluding everything in paths-ignore and we wont trigger the workflow on changes to those directories we dont need this IF statement anymore. we need to just specify the cases